cpp/src/arrow/compute/kernels/aggregate_basic.cc

```diff
@@ -282,6 +282,43 @@ Result<std::unique_ptr<KernelState>> MinMaxInit(KernelContext* ctx,
   return visitor.Create();
 }
 
+// For "min" and "max" functions: dispatch to the MinMax kernel.
+Result<std::unique_ptr<KernelState>> MinOrMaxInit(KernelContext* ctx,
+                                                  const KernelInitArgs& args) {
+  std::vector<ValueDescr> inputs = args.inputs;
+
+  ARROW_ASSIGN_OR_RAISE(auto func, GetFunctionRegistry()->GetFunction("min_max"));
```

Comment:
For the record, it's a bit of a pity to do a second registry lookup when arriving here.